staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresource/strategy.go

```diff
@@ -55,17 +57,21 @@ func NewStrategy(typer runtime.ObjectTyper, namespaceScoped bool, kind schema.Gr
 			kind:            kind,
 			validator:       validator,
 		},
+		converter:      converter,
+		storageVersion: storageVersion,
 	}
 }
 
 func (a customResourceDefinitionStorageStrategy) NamespaceScoped() bool {
 	return a.namespaceScoped
 }
 
-func (customResourceDefinitionStorageStrategy) PrepareForCreate(ctx genericapirequest.Context, obj runtime.Object) {
+func (a customResourceDefinitionStorageStrategy) PrepareForCreate(ctx genericapirequest.Context, obj runtime.Object) {
+	a.converter.ConvertToVersion(obj, schema.GroupVersion{Group: obj.GetObjectKind().GroupVersionKind().Group, Version: a.storageVersion})
```

Comment:
This is not what we do for native resources. The conversion logic is hidden usually in the store configured with RESTOptionGetters. It should not leak into the strategy.

Reply:
I tried that path but it is mostly tailored for native types. There are differences for CRDs that make them unique (e.g. there is no internal type, there is no go type, etc.). It is possible but I am questioning the value of doing it.

Reply:
Also I see conversion to storage version a good “Strategy” “BeforeCreate”ing the object :)

Reply:
I agree conversion belongs in storage, not in strategy